src/Build/BuildCheck/Infrastructure/BuildCheckConnectorLogger.cs

```diff
@@ -60,12 +68,30 @@ private void HandleProjectEvaluationFinishedEvent(ProjectEvaluationFinishedEvent
 
     private void HandleProjectEvaluationStartedEvent(ProjectEvaluationStartedEventArgs eventArgs)
     {
+        if (eventArgs.IsRestore)
+        {
+            isRestore = true;
+            return;
+        }
+
         if (!IsMetaProjFile(eventArgs.ProjectFile))
         {
             _buildCheckManager.StartProjectEvaluation(BuildCheckDataSource.EventArgs, eventArgs.BuildEventContext!, eventArgs.ProjectFile!);
         }
     }
 
+    private void HandleProjectFinishedEvent(ProjectFinishedEventArgs projectFinishedEventArgs)
+    {
+        if (isRestore)
+        {
+            isRestore = false;
```

Comment:
This doesn't make sense to me, won't there be a bunch of ProjectFinished events that will fire one after the other, causing the first one to unset this and the others to do check stuff?

Reply:
Yeah, I figured that out after I ran this for a bit. I had hoped the event was just fired once. Is there an event that is fired at the end of the restore phase that I can use to to control the flow on the connector logger?

Reply:
If we are relying on the restore happen fully before the build - then the first `ProjectEvaluationStartedEventArgs` that is not marked with `IsRestore` denotes the start of the build after the restore.

Reply:
I changed the `isRestore`  reset to happen when we receive another `ProjectEvaluationStartedEventArgs` event.